lib/kapost/bootstrapper.rb

```diff
@@ -92,18 +92,22 @@ def installed?(command)
 
     def right_version?(command, expected_version)
       version, status = cli.capture2e "#{command} --version"
+      if version[0] == 'v'
+        version = version[1..-1]
+      end
       if expected_version[0] == '^'
         next_major = (expected_version[1].to_i + 1).to_s
         Gem::Version.new(version) >= Gem::Version.new(expected_version[1..-1]) && Gem::Version.new(version) < Gem::Version.new(next_major)
       elsif expected_version[0] == "="
         Gem::Version.new(version) == Gem::Version.new(expected_version[1..-1])
+      elsif version.include?("ruby")
```

Comment:
This check feels a little out of place since everything else is only working with versions.  Wondering if we could co-locate all the code that's dealing with different prefixes on the version:

```
def right_version?(command, expected_version)
  version, status = get_command_version(command)

  if expected_version[0] == "^"
    ...
  elsif expected_version[0] == "="
    ...
  else
    ...
  end
end

def get_command_version(command)
  version, status = cli.capture2e "#{command} --version"
  raise CommandError, "Failed to get version for #{command}" unless status.success?

  version = version[1..-1] if version[0] == "v"
  version = version[5..-1] if version =~ "^ruby"
  version
  
end
```

Reply:
@bwebster I agree that it does seem out of place, however, if you are using `rvm` to manage ruby versions, the output of `ruby --version` is not valid SemVer notation (ex: `ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16]`), so `semantic` wouldn't be able to handle that check.
```
ArgumentError: ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16] is not a valid SemVer Version (http://semver.org)
	from /Users/baxter/.rvm/gems/ruby-2.3.1/gems/semantic-1.6.0/lib/semantic/version.rb:14:in `initialize'
	from (irb):3:in `new'
	from (irb):3
	from /Users/baxter/.rvm/rubies/ruby-2.3.1/bin/irb:11:in `<main>'
```

This is why I added the extra conditional for ruby in this loop that does a `.include?` check. 

Can play around with stripping the actual semantic version out of the rvm response for `ruby --version` to make it work.

Reply:
The `get_command_version` in the sample code above would be able to normalize the version to ensure it never starts with `ruby ` or `v`.

Reply:
You are right that it would handle the stripping of `ruby`, but that would leave us with `2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16]`, which still isn't valid semver.

Check out my latest commit. Handles exactly what we are discussing.

Reply:
Although, I'm not sure if it is safe to assume everyone is using RVM. Not sure what the output of `ruby-version` looks like for people who use `rbenv`.....

Reply:
I think that is standard for ruby and is independent of rvm.  I'm using ruby-install and chruby and I get the same output:
```shell
$ ruby --version
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
```